### ISSUE TITLE
Can't get it to work

### ISSUE THREAD

**Issue by felippenardi** [keycastr/keycastr#25]
It was working fine, but suddenly it stopped.

I've already tried to reinstall, check permissions on "Security & Privacy", but it will never display the keys. Can someone help me troubleshoot it?

**Comment by akitchen** [comment]
Have you tried removing it from security and privacy, and re-adding it? Sometimes this is necessary in more recent versions of OS X when keycastr's settings have changed.

**Comment by felippenardi** [comment]
I used to do it every time I used Keycastr—on Yosamite. But it won't do now.

**Comment by akitchen** [comment]
Does the svelte visualizer work for you? Or does the app exit on startup with a 'catastrophic error' dialog displayed?

**Comment by felippenardi** [comment]
Good guess! It doesn't exit on startup anymore since the last time I re-authorized on Security panel, but the svelte indeed works!

The thing is: the only keys svelte will highlight are the ones in the bottom (shift, control, option and command). It doesn't catch "cmd+tab" or "option+n" and so on. Any guess?

**Comment by motiko** [comment]
Same issue here, on Yosemite.

**Comment by motiko** [comment]
I was able to fix it by enlarging the font to maximum and dragging the window into the visible area.
Svelte mode still shows only command buttons (shift alt cmd) .

**Comment by felippenardi** [comment]
Couldnt replicate that. How can I totally uninstall it? It keeps remembering my settings

**Comment by akitchen** [comment]
Where was this build downloaded from?

**Comment by felippenardi** [comment]
Last one was the one in github. But something strange just happened: I closed my mac and when I opened it again, Keycastr started working. It is working perfectly. Can't understand what happened.

I did enlarged the fonts, so when it started working I saw large fonts on my screen. But I did tried before with no success.

Thank you everyone for helping troubleshooting it. 

Should we left this issue open so someone can find and fix the root of the problem?

**Comment by akitchen** [comment]
I have only seen issues like this when using the build available for download on popular internet download sites.

we need to cut a new version and publish it, but unfortunately it is blocked on being able to reflect the version number in the about screen.  Currently that nib is in an unreadable format, making it hard/impossible to edit.  If anybody has any ideas or knows how to convert it, a contribution would be most welcome.

**Comment by akitchen** [comment]
@felippenardi  I ran into this again recently, and tracked it down that the OS wasn't sending keyDown events in some cases, even though it sent modifier events for just the cmd/option/ctrl keys. At the time I had Xcode open, and upon closing Xcode it started working again. Not sure if that's related.

Anyway, this doesn't seem to be a bug in KeyCastr, so I'm closing this. Please try the new version out and let us know if you run into other problems.  Thanks!

**Comment by felippenardi** [comment]
It is still not working... @akitchen How can I delete the settings and anything more Keycastr may store on my computer?

When I delete and open it again, the settings remains persistent.

**Comment by akitchen** [comment]
I would like to separate this into two new issues, as there are two slightly different symptoms I think we understand. Then we can figure out how to fix them.

1) user doesn't see one of the visualizers because of a crazy frame stored in NSUserDefaults; the other visualizer works fine.
* this could maybe happen if the user moved the visualizer around once when on a larger-than-normal display, like a fancy TV or mac with an usual custom screen scale. Try scaling the display and/or increasing the font size to find your visualizer.

2) user doesn't see the default visualizer at all, and the Svelte visualizer only shows modifier keys.
 * The app has stopped receiving keyDown events from the OS, but still receives events when modifier keys change.
 * This is strange because the same event tap registers callbacks for both events, yet one stops firing. Might be a quirk specific to certain versions of OS X (i.e., a bug)

@felippenardi if your svelte visualizer is displaying all keystroke info correctly, then your are in case # 1 above.  Try changing your display scale to find your default visualizer.  Also try adjusting the font all the way up in KeyCastr preferences.  I think we should work on providing a reset mechanism for users in this case.

If your svelte visualizer is only displaying modifier keys, then you are in case # 2 outlined above.  Try closing developer tools or restarting OS X. This one actually doesn't seem to be a KeyCastr bug, believe it or not.

**Comment by liuxueyang** [comment]
Well, it does not work on OS X 10.11.6 as well... even I set the font size to max.

**Comment by akitchen** [comment]
It works fine for me on that same version of OS X. This probably has more
to do with your settings somehow resulting in the keycastr view being
offscreen or not visible.  Sometimes changing the screen resolution can
help as it results in the settings being reset.

**Comment by massisenergy** [comment]
> Have you tried removing it from security and privacy, and re-adding it? Sometimes this is necessary in more recent versions of OS X when keycastr's settings have changed.
> 

**MacOS Sequoia 15.1**, this suggestion worked for me!
